src/sgame/sg_bot_parse.cpp

```diff
@@ -279,32 +279,32 @@ static AIValue_t randomChance( gentity_t*, const AIValue_t* )
 	return AIBoxFloat( random() );
 }
 
-static AIValue_t cvarInt( gentity_t*, const AIValue_t *params )
+// Read a cvar, no matter its type
+static AIValue_t cvar( gentity_t*, const AIValue_t *params )
 {
 	// TODO: improve performance when the need arises
+	// This guess the type of the cvar from what successfully parses.
+	// You may want to change that someday.
 	std::string cvar = AIUnBoxString( params[ 0 ] );
-	int num;
-	if ( !Cvar::ParseCvarValue( Cvar::GetValue( cvar ), num )  )
+	bool boolean;
+	int integer;
+	float floating;
+
+	if ( Cvar::ParseCvarValue( Cvar::GetValue( cvar ), boolean )  )
 	{
-		Log::Warn("could not read cvar '%s' as integer", cvar);
-		num = 0;
+		return AIBoxInt( boolean ? 1 : 0 );
 	}
-
-	return AIBoxInt( num );
-}
-
-static AIValue_t cvarFloat( gentity_t*, const AIValue_t *params )
-{
-	// TODO: improve performance when the need arises
-	std::string cvar = AIUnBoxString( params[ 0 ] );
-	float num;
-	if ( !Cvar::ParseCvarValue( Cvar::GetValue( cvar ), num ) )
+	if ( Cvar::ParseCvarValue( Cvar::GetValue( cvar ), integer )  )
+	{
+		return AIBoxInt( integer );
+	}
+	if ( Cvar::ParseCvarValue( Cvar::GetValue( cvar ), floating )  )
 	{
-		Log::Warn("could not read cvar '%s' as float", cvar);
-		num = 0.0f;
+		return AIBoxFloat( floating );
 	}
 
-	return AIBoxFloat( num );
+	Log::Warn("Bot: could not read cvar '%s'", cvar);
```

Comment:
I suggest appending *as number* to the message. LGTM otherwise

Reply:
Fixed. Merging when pipeline is ready.